### ISSUE TITLE
Not parsing msg.payload properly

### ISSUE THREAD

**Issue by bobolieux** [bartbutenaers/node-red-contrib-msg-router#2]
I am attempting to use the msg router in consistent hash mode to route messages containing IP addresses. Other nodes detect the msg.payload.ip correctly. 

![image](https://user-images.githubusercontent.com/6380326/40201074-0ce5de9a-59ec-11e8-8780-854bc1dd60a0.png)

5/17/2018, 4:05:18 PMnode: 2e7b1626.c9866a
msg.payload.ip : string[12]
"192.168.0.55"

5/17/2018, 4:05:18 PMnode: a697776b.24ba98
msg : string[58]
"The input message doesn't have have a msg.payload.ip field"

**Comment by bartbutenaers** [comment]
Hi @bobolieux ,

Indeed the router node doesn't support **nested properties**.  Good catch ...

I could solve this in two different ways:
1. Check whether the **nested** property exists.  So only generate the error *'The input message doesn't have have a msg.payload.ip field'* when there is no `msg.payload.ip` field.
2. Don't check anymore whether the specified input message field exists.  So for all messages without `msg.payload.ip` field, the IP value will be ***undefined***.  This means all ***undefined*** hashes will be redirected to the same output port (but no error will be generated anymore).

I have currently chosen for option 2.  Or do you think option 1 might be better for some reason?

You can test option 2 by installing this node directly from GitHub:
`npm install bartbutenaers/node-red-contrib-msg-router`

Please let me know if it fits your needs, so I can publish the new version **0.0.4** to NPM.

Kind regards,
Bart Butenaers

**Comment by bartbutenaers** [comment]
Hi @bobolieux,

My previous answer was a bit too quick.  Cannot do it that way, because I don't want existing flows to be broken: those flows expect that messages are ignored if the input message doesn't contain the key field.

So I have added following checkbox:
![image](https://user-images.githubusercontent.com/14224149/40268079-760c1eb2-5b67-11e8-8b58-9aef03afe5bb.png)

With following help in the info tab:
![image](https://user-images.githubusercontent.com/14224149/40268086-8e460f38-5b67-11e8-8342-505dc794ff25.png)

And in the readme file:
![image](https://user-images.githubusercontent.com/14224149/40268093-a338d074-5b67-11e8-8b32-d49ab2160594.png)

Here is my test flow:
![image](https://user-images.githubusercontent.com/14224149/40268096-b190203c-5b67-11e8-959f-66a9b28bce00.png)
```
[{"id":"6ce11ea4.92b82","type":"msg-router","z":"dbde7fab.e1c62","routerType":"hashing","topicDependent":false,"counterReset":false,"msgKeyField":"payload.ip","undefinedHash":true,"outputsInfo":[{"active":true,"clone":false,"delay":"0","weight":"0"},{"active":true,"clone":false,"delay":"0","weight":"0"},{"active":true,"clone":false,"delay":"0","weight":"0"}],"name":"","delaying":"unrelated","msgControl":false,"outputs":3,"x":710,"y":280,"wires":[["bf9f35e2.a72ea8"],["2bf1d6bb.7ae99a"],["941d1a99.ab42d8"]]},{"id":"d6d96484.f28578","type":"inject","z":"dbde7fab.e1c62","name":"Inject msg with 'payloay.ip' field","topic":"","payload":"","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":230,"y":280,"wires":[["a60de27f.59c92"]]},{"id":"bf9f35e2.a72ea8","type":"debug","z":"dbde7fab.e1c62","name":"Output 1","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","x":920,"y":240,"wires":[]},{"id":"2bf1d6bb.7ae99a","type":"debug","z":"dbde7fab.e1c62","name":"Output 2","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","x":920,"y":280,"wires":[]},{"id":"941d1a99.ab42d8","type":"debug","z":"dbde7fab.e1c62","name":"Output 3","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","x":920,"y":320,"wires":[]},{"id":"a60de27f.59c92","type":"function","z":"dbde7fab.e1c62","name":"Set msg.payload.ip","func":"return {payload: {ip: '192.168.0.55'}};","outputs":1,"noerr":0,"x":490,"y":280,"wires":[["6ce11ea4.92b82"]]},{"id":"dfb69c6a.9581b","type":"inject","z":"dbde7fab.e1c62","name":"Inject msg without 'payloay.ip' field","topic":"","payload":"","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":440,"y":340,"wires":[["6ce11ea4.92b82"]]}]
```

When the checkbox is unselected (like in existing flows) there will be an error if `msg.payload.ip` doesn't exist in the input message.  And when selected, all messages with ***undefined*** hash will be forwarded to the same output (in this case Output 2):
![image](https://user-images.githubusercontent.com/14224149/40268107-1b7944ec-5b68-11e8-99fc-d2b2d928ade4.png)

Would be nice if you could let me know it this works as expected.
Bart

**Comment by bartbutenaers** [comment]
This functionality is now available on NPM in version **0.0.4**.
